pallets/capacity/src/lib.rs

```diff
@@ -601,6 +613,23 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Claim all outstanding rewards earned from ProviderBoosting.
+		#[pallet::call_index(6)]
+		#[pallet::weight(T::WeightInfo::claim_staking_rewards())]
+		pub fn claim_staking_rewards(origin: OriginFor<T>) -> DispatchResult {
+			let staker = ensure_signed(origin)?;
+			ensure!(
+				ProviderBoostHistories::<T>::contains_key(staker.clone()),
+				Error::<T>::NotAProviderBoostAccount
+			);
+			let total_to_mint = Self::do_claim_rewards(&staker)?;
+			Self::deposit_event(Event::ProviderBoostRewardClaimed {
+				account: staker.clone(),
```

Comment:
Can you please check if the clone can be removed from here. It seems at first sight that we might not need to do a clone.